### PR TITLE
Add data-testid to job-related components

### PR DESCRIPTION
**Pull request by marlovr**
Why:
* Cypress tests in xp-registration involving these components require this attribute

**Release Type:** *Dev Improvement* 

Fixes https://x-team-internal.atlassian.net/browse/XP-3689

## Description

* Cypress best practices require the use of the `data-testid` attribute, rather than `className` or `id`
* The components required to run the View Jobs Board E2E have had the `data-testid` attribute added to them in accordance with previously established naming conventions.
* The affected components are listed below in "Impacted areas of application"

## Checklist

**Before submitting a Pull Request,** please make sure the following is done:

- [x ] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `PEER REVIEW` column
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`) (**NOTE:** This is temporarily disabled. Skip it.)
- [x] check cleanup tasks (https://github.com/x-team/xp/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [x] if added a new UI component that is meant to be used directly by other apps, remember to export that from the appropriate bundle file: `src/index.js` and/or `src/registration.js`
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases and/or debug cases
- [x] if you've fixed a bug, make sure to also include a new story that will expose declarations with problematic data that caused the bug in the first place, so that we can assure that no regressions will pop up on future interactions
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot (**NOTE:** This is temporarily disabled. Skip it.)
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Related PRs

branch | PR
------ | ------

XP-3623-E2E-View-Jobs-Board | [XP Registration](https://github.com/x-team/xp-registration/tree/XP-3623-E2E-View-Jobs-Board)

## Steps to Test or Reproduce

N/A

## Impacted Areas in Application

`src/components/ui/ApplicantScreenNotification.js`
`src/components/ui/HeaderBar.js`
`src/components/ui/JobApplicationCard.js`
`src/components/ui/JobCard.js`
`src/components/ui/Modal.js`
`src/components/ui/WelcomeHero.js`
`src/components/ui/EmptyState.js`

### CODE REVIEW

**Comment by nicksp** [comment]
Hey @marlovr Could you please resolve those merge conflicts before we take it further?

**Comment by nicksp** [comment]
Hey @marlovr Please rebase with `develop` by keeping in mind changes from here https://github.com/x-team/xp-ui/pull/532 where we basically removed `/lib/*` files from the repo, gitignored then and we auto generate them now on a consumer side when installing `xp-ui`. This makes the process less error-prone and no need to generate manually anymore.

**Comment by marlovr** [comment]
Hey @nicksp I've rebased onto develop :) Hopefully it will be all good now!

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,7 @@ npm-debug.log
 .cache
 .vscode
 .cache
+
+# production
+/lib
```
Comment (nicksp) [inline .gitignore:10]: Is this a new change? Same goes for other things that seemed to be merged as part of https://github.com/x-team/xp-ui/pull/532

Comment (marlovr) [reply .gitignore:10]: I think I've messed up the rebase. I think the best option for me is to create a new PR off of the latest `develop` and include the changes then as they're really trivial ones. What is the process for closing 'dead' PRs?

**Comment by marlovr** [comment]
Closed as duplicate of [XP-3689-Add-data-testid-attribute-to-job-related-components]()

File: .gitignore (hunk 1 above)
Comment (nicksp) [reply .gitignore:10]: > What is the process for closing 'dead' PRs?

Just close the PR **without** merging it and remove the remote branch. Should be sufficient 😄

**Pull request closed without merging**